src/inventorymanager.cpp

```diff
@@ -154,6 +154,91 @@ IMoveAction::IMoveAction(std::istream &is, bool somewhere) :
 	}
 }
 
+void IMoveAction::swapDirections()
+{
+	std::swap(from_inv, to_inv);
+	std::swap(from_list, to_list);
+	std::swap(from_i, to_i);
+}
+
+const void IMoveAction::onPutAndOnTake(const ItemStack &src_item, ServerActiveObject *player)
+{
+	ServerScripting *sa = PLAYER_TO_SA(player);
+	if (to_inv.type == InventoryLocation::DETACHED)
+		sa->detached_inventory_OnPut(*this, src_item, player);
+	else if (to_inv.type == InventoryLocation::NODEMETA)
+		sa->nodemeta_inventory_OnPut(*this, src_item, player);
+	else {
+		assert(to_inv.type == InventoryLocation::PLAYER);
```

Comment:
I prefer an else if here and putting `assert(false);` into an else branch

Reply:
Asserts are dummied in non-debug builds. If that's not what the author intends, they might want to look into `FATAL_ERROR()` instead, which is a persistent variant of `assert`.

@appgurueu

Reply:
> I prefer an else if here and putting assert(false); into an else branch
I'm tempted to change it as this would have all branches be one-liners.

> Asserts are dummied in non-debug builds
Hmm. I'd prefer non-debug builds not to crash.